### PR TITLE
[MODULAR, PROBABLY-ISH] fixes funny grab interactions (headslams/suplexes/dislocations)

### PR DESCRIPTION
**Pull request by Hatterhat**
## About The Pull Request
image

fixes funny grab interactions

puts all grab interactions on RMB, pretty much, keeping combat mode LMBs free for kicking people in the face or whatever

## How This Contributes To The Skyrat Roleplay Experience
i love hurting people

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
image

</details>

## Changelog

:cl:
fix: Grab interactions (slamming heads into floors, suplexes(??), dislocations) are fixed and are all now on RMB when aggressive grabbing (or above) someone.
/:cl:

### CODE REVIEW

**Comment by PCPort** [comment]
SPEEDMERGE

File: code/__DEFINES/~skyrat_defines/combat.dm [hunk 1]
```diff
@@ -201,4 +175,34 @@
 		user.changeNext_move(CLICK_CD_MELEE)
 	return
 
+/datum/species/proc/try_headslam(mob/living/carbon/human/user, mob/living/carbon/human/target, obj/item/bodypart/affecting)
```
Comment (Zonespace27) [inline code/__DEFINES/~skyrat_defines/combat.dm:179]: codedoc

File: code/__DEFINES/~skyrat_defines/combat.dm [hunk 2]
```diff
@@ -201,4 +175,34 @@
...
+		to_chat(user, span_danger("You slam [target.name] head against the floor!"))
+
+		//Check to see if our head is protected by at least 20 melee armor
+		if(head_knock)
+			target.adjustOrganLoss(ORGAN_SLOT_BRAIN, 15)
+		var/fun_times_at_the_headbash_factory = (head_knock ? 8 : 3) // arbitrary numbers
+		target.apply_damage(15, BRUTE, affecting, armor_block, wound_bonus = fun_times_at_the_headbash_factory, bare_wound_bonus = fun_times_at_the_headbash_factory)
+		playsound(target, 'sound/effects/hit_kick.ogg', 70)
+		log_combat(user, target, "headsmashes", "against the floor")
+		return
```
Comment (Zonespace27) [inline code/__DEFINES/~skyrat_defines/combat.dm:191]: return does nothing

File: code/__DEFINES/~skyrat_defines/combat.dm [hunk 3]
```diff
@@ -201,4 +175,34 @@
...
+	if(target.stat != CONSCIOUS)
+		time_doing = 2 SECONDS
+		target.visible_message(span_danger("[user.name] holds [target.name]'s head tightly..."), ignored_mobs=user)
+		to_chat(user, span_danger("You grasp [target.name]'s head to prepare to slam it down..."))
+	else
+		target.visible_message(span_danger("[user.name] holds [target.name]'s head and tries to overpower [target.p_them()]!"), \
+			span_userdanger("You struggle as [user.name] holds your head and tries to overpower you!"), ignored_mobs=user)
+		to_chat(user, span_danger("You grasp [target.name]'s head and try to overpower [target.p_them()]..."))
+	user.changeNext_move(time_doing)
+	if(do_after(user, time_doing, target))
```
Comment (Zonespace27) [inline code/__DEFINES/~skyrat_defines/combat.dm]: early return

File: code/__DEFINES/~skyrat_defines/combat.dm [hunk 4]
```diff
@@ -201,4 +175,34 @@
 		user.changeNext_move(CLICK_CD_MELEE)
 	return
 
+/datum/species/proc/try_headslam(mob/living/carbon/human/user, mob/living/carbon/human/target, obj/item/bodypart/affecting)
+	var/time_doing = 4 SECONDS
+	if(target.stat != CONSCIOUS)
+		time_doing = 2 SECONDS
+		target.visible_message(span_danger("[user.name] holds [target.name]'s head tightly..."), ignored_mobs=user)
```
Comment (Zonespace27) [inline code/__DEFINES/~skyrat_defines/combat.dm]: `.name` is implicit when doing [ref to atom]

Comment (Zonespace27) [inline code/__DEFINES/~skyrat_defines/combat.dm]:
```suggestion
		target.visible_message(span_danger("[user.name] holds [target.name]'s head tightly..."), ignored_mobs = user)
```

File: code/__DEFINES/~skyrat_defines/combat.dm [hunk 5]
```diff
@@ -201,4 +175,34 @@
...
+		to_chat(user, span_danger("You grasp [target.name]'s head to prepare to slam it down..."))
+	else
+		target.visible_message(span_danger("[user.name] holds [target.name]'s head and tries to overpower [target.p_them()]!"), \
+			span_userdanger("You struggle as [user.name] holds your head and tries to overpower you!"), ignored_mobs=user)
```
Comment (Zonespace27) [inline code/__DEFINES/~skyrat_defines/combat.dm]:
```suggestion
			span_userdanger("You struggle as [user.name] holds your head and tries to overpower you!"), ignored_mobs = user)
```

**Review by Zonespace27 — CHANGES_REQUESTED** [review]
Few small nits, thanks for fixing this!

File: code/__DEFINES/~skyrat_defines/combat.dm [hunk 6]
```diff
@@ -201,4 +175,36 @@
...
+	if(!do_after(user, time_doing, target))
+		return
+	var/armor_block = target.run_armor_check(affecting, MELEE)
+	var/head_knock = FALSE
+	// Check to see if our head is protected by at least HEADSMASH_BLOCK_ARMOR (20 melee armor)
+	if(armor_block < HEADSMASH_BLOCK_ARMOR)
+		head_knock = TRUE
+
+	target.visible_message(span_danger("[user.name] violently slams [target.name]'s head into the floor!"), \
+		span_userdanger("[user.name] slams your head against the floor!"), ignored_mobs=user)
```
Comment (Zonespace27) [inline code/__DEFINES/~skyrat_defines/combat.dm]:
```suggestion
		span_userdanger("[user.name] slams your head against the floor!"), ignored_mobs = user)
```

**Review by Zonespace27 — APPROVED** [review]

**Pull request merged**